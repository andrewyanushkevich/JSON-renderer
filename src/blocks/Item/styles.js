import styled from 'styled-components';

export const Title = styled.h2`
  text-align: center;
`;

export const StyledItem = styled.div`
  width: 60%;
  height: 30%;
  margin: 10%;
  & > div {
    display: flex;
    align-items: space-between;
  }
`;

export const Img = styled.img`
  width: 20%;
  height: 20%;
`;
