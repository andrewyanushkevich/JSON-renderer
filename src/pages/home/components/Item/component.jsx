/* eslint-disable no-underscore-dangle */
/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import Rating from 'blocks/Rating';
import { Tag } from 'antd';

import {
  StyledItem,
  Title,
  TagsWrapper,
  RatingWrapper,
  Img,
  Price,
} from './styles';

class Item extends Component {
  handleViewFullItem = () => {
    const { item, history } = this.props;
    const id = item._id.$oid;
    history.push(`/${id}`);
  };

  render() {
    const { item } = this.props;
    return item ? (
      <StyledItem onClick={this.handleViewFullItem}>
        <Title>{item.title}</Title>
        <div>
          <div>
            <Img srcSet={item.images[0]} />
            <TagsWrapper>
              {item.color.map(elem => (
                <Tag color={elem}>{elem}</Tag>
              ))}
            </TagsWrapper>
            <RatingWrapper>
              <Rating stars={item.rating} totalStars={5} />
            </RatingWrapper>
          </div>
          <div>
            <Price>{item.price}$</Price>
          </div>
        </div>
      </StyledItem>
    ) : null;
  }
}

Item.propTypes = {
  item: PropTypes.shape({
    _id: PropTypes.shape({
      $oid: PropTypes.string,
    }),
    title: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    size: PropTypes.arrayOf(PropTypes.string).isRequired,
    images: PropTypes.arrayOf(PropTypes.string).isRequired,
    rating: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired,
    color: PropTypes.arrayOf(PropTypes.string).isRequired,
    tags: PropTypes.arrayOf(PropTypes.string).isRequired,
  }).isRequired,
};

export default withRouter(Item);
