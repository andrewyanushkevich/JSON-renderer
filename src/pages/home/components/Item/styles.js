/* istanbul ignore file */
import styled from 'styled-components';
import { Tag } from 'antd';

export const Title = styled.h2`
  justify-content: center;
  font-size: 20px;
  padding: 10px;
`;

export const StyledItem = styled.div`
  & > div,
  h2 {
    display: flex;
    flex-direction: row;
    & > div {
      display: flex;
      flex-direction: column;
    }
  }
  display: flex;
  flex-direction: column;
  flex-basis: 30%;
  width: 300px;
  border-radius: 10px;
  margin: 20px 0px;

  background-color: snow;

  :hover {
    cursor: pointer;
    box-shadow: 0px 0px 28px -7px rgba(0, 0, 0, 0.75);
    transition: 1s;
  }

  @media only screen and (max-width: 576px) {
    width: 80%;
  }
`;

export const Img = styled.img`
  width: 120px;
  height: 160px;
  margin: 0 auto;
`;

export const TagsWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  flex-wrap: wrap;
  width: 100%;
  margin: 20px;
`;

export const RatingWrapper = styled.div`
  width: 200px;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: auto;
`;

export const Price = styled.em`
  font-size: 18px;
  width: 50px;
`;

export const TagItem = styled(Tag)`
  margin: 5px;
  color: black;
`;
