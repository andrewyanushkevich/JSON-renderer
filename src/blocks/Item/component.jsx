import React, { Component } from "react";

import { StyledItem, Title, Img } from './styles';

class Item extends Component {
  render() {
    const { item } = this.props;
    return item ? (
      <StyledItem>
        <Title>{item.title}</Title>
        <div>
          <Img srcSet={item.images[0]}></Img>
          <p>{item.description}</p>
        </div>
        <span>{item.price}</span>
        <span>{item.rating}</span>
      </StyledItem>
    ): null;
  }
}

export default Item;
