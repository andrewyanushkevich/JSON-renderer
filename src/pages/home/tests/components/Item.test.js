import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import Item from 'pages/home/components/Item';

describe('(Component) Item', () => {
  let component;
  const id = '5bb6b6936cfa75c4e50bac5e';
  const clickFn = jest.fn();
  const props = {
    item: {
      _id: { $oid: '5bb6b6936cfa75c4e50bac5e' },
      title: 'Polo red',
      price: 50,
      size: ['XL', 'L', 'M', 'S', 'XS'],
      images: [
        'https://gloimg.samcdn.com/S/pdm-product-pic/Clothing/2016/06/03/source-img/20160603190149_72043.jpg',
      ],
      rating: 4,
      description: 'uspeenatis magna. Sed pulvinar nunc quam, nec port tit',
      tags: ['Polos'],
      color: ['red'],
    },
  };
  beforeEach(() => {
    component = mount(
      <MemoryRouter>
        <Item {...props} onClick={clickFn} />
      </MemoryRouter>,
    );
  });
  it('should render correctly', () => {
    const output = renderer.create(
      <MemoryRouter>
        <Item {...props} onClick={clickFn} />
      </MemoryRouter>,
    );
    expect(output).toMatchSnapshot();
  });

  it('Should change location.pathname', () => {
    expect(component.find('Item').props().location.pathname).toBe('/');

    component
      .find('div')
      .first()
      .simulate('click');

    expect(component.find('Item').props().location.pathname).toBe(`/${id}`);
  });
});
