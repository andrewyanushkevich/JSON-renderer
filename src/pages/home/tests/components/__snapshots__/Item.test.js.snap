// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Item should render correctly 1`] = `
<div
  className="sc-bwzfXH drhLpv"
  onClick={[Function]}
>
  <h2
    className="sc-bdVaJa kbFZZC"
  >
    Polo red
  </h2>
  <div>
    <div>
      <img
        className="sc-htpNat kRFHUR"
        srcSet="https://gloimg.samcdn.com/S/pdm-product-pic/Clothing/2016/06/03/source-img/20160603190149_72043.jpg"
      />
      <div
        className="sc-ifAKCX kYHPDS"
      >
        <div>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 "
            color="orange"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 "
            color="orange"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 "
            color="orange"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 "
            color="orange"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 "
            color="lightgray"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
      </div>
      <div
        className="sc-bxivhb eUsJrq"
      >
        <span
          className="ant-tag ant-tag-red sc-bZQynM didVjt"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        >
          red
        </span>
      </div>
    </div>
    <div>
      <em
        className="sc-EHOje hvCOeX"
      >
        50
        $
      </em>
      <div>
        <span
          className="ant-tag sc-bZQynM didVjt"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        >
          XL
        </span>
        <span
          className="ant-tag sc-bZQynM didVjt"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        >
          L
        </span>
        <span
          className="ant-tag sc-bZQynM didVjt"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        >
          M
        </span>
        <span
          className="ant-tag sc-bZQynM didVjt"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        >
          S
        </span>
        <span
          className="ant-tag sc-bZQynM didVjt"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        >
          XS
        </span>
      </div>
    </div>
  </div>
</div>
`;
